core/dbt/contracts/graph/nodes.py

```diff
@@ -1411,6 +1423,28 @@ class Group(BaseNode):
     resource_type: NodeType = field(metadata={"restrict": [NodeType.Group]})
 
 
+# ====================================
+# SemanticModel and related classes
+# ====================================
+
+
+@dataclass
+class NodeRelation(dbtClassMixin):
+    alias: str
+    schema_name: str  # TODO: Could this be called simply "schema" so we could reuse StateRelation?
+    database: Optional[str] = None
```

Comment:
From [this comment](https://github.com/dbt-labs/dbt-core/issues/7503#issuecomment-1538583101) -- does this still need to include a unified relation_name that respects quoting + include policies?

Reply:
if we do -- we'll probably need the adapter.Relation to do something like this during compilation:
```python
adapter = get_adapter(self.config)
relation_cls = adapter.Relation
relation_name = str(relation_cls.create_from(self.config, node))
```
(from https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/compilation.py#L486-L488)

Reply:
I don't think the SemanticModels actually get compiled, since they're yaml-only. There is a question of whether they need the individual pieces (identifier/schema/database) or just the relation_name...

Reply:
@QMalcolm I'm not sure of the answer to this, but I suspect the answer may be yes. What makes sense from the perspective of MetricFlow integration?

Reply:
@MichelleArk Yep! We need to include the unified relation_name, and it's desirable that it respects quoting + include policies.

As for whether we need both the individual pieces as well as the `relation_name`, for this PR yes, long term we don't have to. MetricFlow is fine with having the parts or the relation_name. It was suggested that we use this node_relation object as it was a pattern that was already used elsewhere, thus the shape of the object currently. Though MetricFlow currently uses the `relation_name` attribute as seen [here](https://github.com/dbt-labs/metricflow/blob/4e64f1656382a9b2001e597681823f0b0cf751a3/metricflow/dataset/convert_semantic_model.py#L407), and ignores the individual properties. However, the individual properties as well as the `relation_name` are currently required by the protocol. So for the scope of this PR, we should include the individual properties as well as the `relation_name`. Otherwise we'd first need to update and release a new DSI (which would include a number of other schema changes that have been made), and then propagate all those changes in core. Although the `node_relation` changes might be trivial in core, propagating the other changes with the new DSI could eat a fair bit of time.